File: VideoDemo/app/src/main/java/com/dilip/videodemo/SecondActivity.java
package com.dilip.videodemo;

import android.net.Uri;
import android.os.Bundle;
import android.widget.MediaController;
import android.widget.VideoView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class SecondActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);

        // Initialize the VideoView for online video playback
        VideoView videoView = findViewById(R.id.onlineVideoView);

        // Set the path for the online video resource
        String onlinePath = "https://file-examples.com/storage/fef545ae0b661d470abe676/2017/04/file_example_MP4_480_1_5MG.mp4";
        Uri onlineVideoUri = Uri.parse(onlinePath);

        // Set the video URI and start playback
        videoView.setVideoURI(onlineVideoUri);
        videoView.start();

        // Create a media controller for playback controls
        MediaController mediaController = new MediaController(this);
        videoView.setMediaController(mediaController);
        mediaController.setAnchorView(videoView);
    }
}

File: ActivityCycleDemo/app/src/main/java/com/dilip/myapplication/SecondActivity.kt
package com.dilip.myapplication

import android.os.Bundle
import android.util.Log
import android.widget.TextView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class SecondActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_second)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
        Log.i("MYTAG", "SecondActivity: OnCreate")
        val userName = intent.getStringExtra("USER")
        val textView = findViewById<TextView>(R.id.tvOffers)
        val message = "$userName, you will get one month free subscription for free."
        textView.text = message
    }

    override fun onStart() {
        super.onStart()
        Log.i("MYTAG", "SecondActivity: OnStart")
    }

    override fun onResume() {
        super.onResume()
        Log.i("MYTAG", "SecondActivity: OnResume")
    }

    override fun onPause() {
        super.onPause()
        Log.i("MYTAG", "SecondActivity: OnPause")
    }

    override fun onStop() {
        super.onStop()
        Log.i("MYTAG", "SecondActivity: OnStop")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.i("MYTAG", "SecondActivity: OnDestroy")
    }

    override fun onRestart() {
        super.onRestart()
        Log.i("MYTAG", "SecondActivity: OnRestart")
    }

}